app/src/main/java/io/github/droidkaigi/confsched2018/presentation/sessions/SessionsViewModel.kt

```diff
@@ -23,13 +23,31 @@ class SessionsViewModel @Inject constructor(
         private val repository: SessionRepository,
         private val schedulerProvider: SchedulerProvider
 ) : ViewModel(), LifecycleObserver {
-    val rooms: LiveData<Result<List<Room>>> by lazy {
+    private val tabModeLiveData = MutableLiveData<SessionTabMode>().apply {
+        postValue(SessionTabMode.Room)
+    }
+    private val rooms: LiveData<Result<SessionTab>> by lazy {
         repository.rooms
+                .map { SessionTab.Room(it) as SessionTab }
+                .toResult(schedulerProvider)
+                .toLiveData()
+    }
+    private val schedules: LiveData<Result<SessionTab>> by lazy {
+        repository.schedules
+                .map { SessionTab.Schedule(it) as SessionTab }
                 .toResult(schedulerProvider)
                 .toLiveData()
     }
+    val tabMode: SessionTabMode
+        get() = tabModeLiveData.value ?: throw IllegalStateException("null is not allowed")
```

Comment:
I know that this will be a discussion, but I think that "!!" should be used. 🤔 
Because If we do it, we should use always `?: throw IllegalStateException("null is not allowed")` when using the Android framework nullable. 🤔 
What do you think?

Reply:
It just depends on my prefer. 😇 Generally speaking, force-unwrap is not so good, and I think so too. Because I believe sometimes we cannot know the true possibility of nullable if `force-unwrap` is used. 

For instance, there might be a writer who uses force-unwrap in early phase of his/her changes, for quick development. It also makes sense for me if `null` is one of rare cases.
However, if he/she might forgot to add an error case and send a patch, then we don't have a way to know whether or not he/she forgot that. (Of course, it would be solved by asking him/her)
It's one of reasons why I don't like force-unwrap.

But, in the case of using the Android framework nullable, I think it has an another aspect. Lifecycle-aware nullability is one of biggest examples. 
In a non-UI thread, we shouldn't use force-unwrap because it's actually nullable. However, it's okay to use force-unwrap, for example, in an observer of LiveData in Activity/Fragment, right?

So... I'd like to follow the current strategy. Which of force-unwrap or explicit exceptions do you usually use?

Reply:
I believe there is not much difference between "!!" and throwing exceptions. 
I use "!!" for what I think is not null and default values ​​for early return and Elvis operator for things that might be null.

Reply:
But if it can be null and it seems to be due to a coding mistake it is good to explicitly write throw like now. Because messages and specific class can be used.

Reply:
I did not know that the value of this livedata can actually be null, so I commented.
I think that it is OK as it is.